src/util.ts

```diff
@@ -1,40 +1,46 @@
-import { TextDelimiterStream } from "https://deno.land/std@0.189.0/streams/mod.ts";
+import { TextDelimiterStream } from "https://deno.land/std@0.189.0/streams/text_delimiter_stream.ts";
+import { ModelError, RequiredError } from "./types.ts";
 
-export function throwError(
-  data: { error?: { type: string; message: string; code: string } },
-) {
-  if (data.error) {
-    let errorMessage = `${data.error.type}`;
-    if (data.error.message) {
-      errorMessage += ": " + data.error.message;
-    }
-    if (data.error.code) {
-      errorMessage += ` (${data.error.code})`;
-    }
-    // console.log(data.error);
-    throw new Error(errorMessage);
+export function throwErrorIfNeeded(response: unknown) {
```

Comment:
Could you split this off into another PR? Ref: #18

Reply:
Should we restore the function code and create a new PR with this code and the issue reference?

I don't wanna create a branch from this branch. Should i wait until this be merged?

Sorry, I'm not an expert on github. I don't know the best practices for the prs T-T

Reply:
> Should we restore the function code and create a new PR with this code and the issue reference?

Yes, this would be great!